src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs

```diff
@@ -144,133 +150,156 @@ public static IHostBuilder CreateHostBuilder(IConsole console, string[] urls, st
 
                     builder.AddKeyPerFileWithChangeTokenSupport(path, optional: true, reloadOnChange: true);
                     builder.AddEnvironmentVariables(ConfigPrefix);
+
+                    if (!noAuth && tempApiKey)
```

Comment:
nit: consider using authMode since you've already worked out exactly what mode you are in.